### PR TITLE
Added example on how to do open/close a port

### PR DESCRIPTION
**Pull request by tree786**
I have added a small code sample to show how to automate the process of port forwarding (and closing after you are done).
Thanks to @mhilbrunner for explaining how to achieve this (see https://github.com/godotengine/godot/pull/18780)

### CODE REVIEW

**Comment by Anutrix** [comment]
Why was this closed?

**Comment by tree786** [comment]
I can open a new one, but I don't understand why I'm getting a travis error, it says 'Some checks were not successful' (I'm new to making PR's in github)

File: modules/upnp/doc_classes/UPNP.xml [hunk 1]
```diff
@@ -5,6 +5,20 @@
 	</brief_description>
 	<description>
 		Provides UPNP functionality to discover [UPNPDevice]s on the local network and execute commands on them, like managing port mappings (port forwarding) and querying the local and remote network IP address. Note that methods on this class are synchronous and block the calling thread.
+		To forward a specific port:
+		[codeblock]
+		const port=7777 # Specify the port you wish to forward
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: Style issue: constants are `UPPER_CASE`, and there should be spaces around operators.

File: modules/upnp/doc_classes/UPNP.xml [hunk 2]
```diff
@@ -5,6 +5,20 @@
...
+		var upnp=UPNP.new() # Create a UPNP class
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: Spaces around operators. The comment does not seem useful, it's pretty clear that this *instantiates* a `UPNP` class.

File: modules/upnp/doc_classes/UPNP.xml [hunk 3]
```diff
@@ -5,6 +5,20 @@
...
+		upnp.discover(2000,2,"InternetGatewayDevice")
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: Spaces after commas.

File: modules/upnp/doc_classes/UPNP.xml [hunk 4]
```diff
@@ -5,6 +5,20 @@
 	</brief_description>
 	<description>
 		Provides UPNP functionality to discover [UPNPDevice]s on the local network and execute commands on them, like managing port mappings (port forwarding) and querying the local and remote network IP address. Note that methods on this class are synchronous and block the calling thread.
+		To forward a specific port:
+		[codeblock]
+		const port=7777 # Specify the port you wish to forward
+		var upnp=UPNP.new() # Create a UPNP class
+		upnp.discover(2000,2,"InternetGatewayDevice") 
+		upnp.add_port_mapping(port) # This line does the forwarding
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: The comment is not particularly useful, this repeats what the example's instruction already said.

File: modules/upnp/doc_classes/UPNP.xml [hunk 5]
```diff
@@ -5,6 +5,20 @@
...
 	<description>
 		Provides UPNP functionality to discover [UPNPDevice]s on the local network and execute commands on them, like managing port mappings (port forwarding) and querying the local and remote network IP address. Note that methods on this class are synchronous and block the calling thread.
+		To forward a specific port:
+		[codeblock]
+		const port=7777 # Specify the port you wish to forward
+		var upnp=UPNP.new() # Create a UPNP class
+		upnp.discover(2000,2,"InternetGatewayDevice") 
+		upnp.add_port_mapping(port) # This line does the forwarding 
+		[/codeblock]
+
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: There shouldn't be empty lines in the XML, they will be parsed as separate paragraphs.

File: modules/upnp/doc_classes/UPNP.xml [hunk 6]
```diff
@@ -5,6 +5,20 @@
...
+		[codeblock]
+		const port=7777 # Specify the port you wish to forward
+		var upnp=UPNP.new() # Create a UPNP class
+		upnp.discover(2000,2,"InternetGatewayDevice") 
+		upnp.add_port_mapping(port) # This line does the forwarding 
+		[/codeblock]
+		
+		To close a specific port (e.g. after you have finished with it):
+		[codeblock]
+		upnp.delete_port_mapping(port) # This line closes the port
```
Comment (akien-mga) [inline modules/upnp/doc_classes/UPNP.xml]: Comment also not useful.

**Comment by akien-mga** [comment]
The added example seems good, but there are style issues. See https://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_styleguide.html

Could you also squash the commits together?

**Comment by tree786** [comment]
Thanks! I had no idea about these issues. I don't know how to update the PR with the new changes. Should I close this and open a new one?

**Comment by akien-mga** [comment]
You can rebase the branch following instructions on http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request to update this PR.

**Comment by akien-mga** [comment]
Merged manually as 2a6385372 with the requested changes

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request closed without merging**